### DISCUSSION TITLE
Creating non-RDBMS plugin

### DISCUSSION THREAD

**Discussion by Canadug** [dbeaver/dbeaver#14313] [DBeaver Development]
Hi all, I've used dbeaver a bunch in the past for SQL Server.  I'm interested in knowing:

1. Can anyone create a new plugin for dbeaver?  Is there documentation on crafting a new one?
2. Can a plugin be made for a non-RDBMS proprietary database which is neither JDBC nor ODBC compliant? I have no issue writing my communication adapter.
3. If this is all possible, how much of an up-hill climb could this be?

Thanks all!

**Comment by LonwoLonwo** [comment]
Hello @Canadug 

1. You can create a new plugin for DBeaver, yes. In your fork, as an example. Also, you can contribute your plugin to DBeaver and we can merge it into our project if it will be good. Such cases have been in DBeaver history :) And sorry, but we do not have such kind of documentation. Only source code.
2. You can try, but it will be very hard. Communication with databases in the project occurs through drivers that are usually JDBC or ODBC.

**Comment by hanxuanliang** [comment]
Hello. Recently, I have been developing a dbeaver plugin for a new database, but I am a bit confused about Eclipse and am not sure whether to use a plugin project or a Maven project. I was thinking, could the official team possibly release a document to guide new developers on initializing a plugin project? This is just my suggestion.

**Reply by LonwoLonwo** [reply]
Hello @hanxuanliang 

Maybe this wiki page will be helpful for you https://github.com/dbeaver/dbeaver/wiki/Develop-in-Eclipse
